src/tests/utils/fakeit.hpp

```diff
@@ -0,0 +1,12162 @@
+#pragma once
```

Comment:
Is there a conan package for `fakeit`?

Reply:
Maybe, but I'm done some changes into it

Reply:
added another mocking library

Reply:
Fyi gmock is also handy to mock grpc/protobuf they have an option to directly generate mocks when compiling protos